### ISSUE TITLE
Minimize to system tray, update icon with total players online

### ISSUE THREAD

**Issue by Rushing** [Deceth/Battle-City#10]
- Ability to minimize Battle City in full screen or windowed mode
- When minimized, show Battle City icon in system tray
- Update icon in system tray to show total players online

> Rushing said: 
> 
> I know you aren't really interested in this project anymore Weebo, but if you had this small feature... well, just read;
> "This is the 16th. Once the 26th comes unless someone steps up to the plate BC will go down for the count.
> 
> BC has two problems right now. First the server is about to be shut off. That problem is easy to fix.
> 
> Second problem is that it is hard to retain players. That problem we haven't been able to fix in a long time.
> 
> I have made a suggestion about a programing change to BC. The change is reading the population count which is already displayed and then sending that information out to the players as a pop up program or preferably as an Instant Message. I can not program so I will have to leave that to someone who can. I give my word that if someone create the program of which I desire a few things will happen. First I truly hope that the player count will rise as people will know when people are playing and that would also encourage/remind them to play. Second I give my word that if that is created for your efforts I will personally sponsor the battle city server for six months.
> 
> All those who read this, you decide. The countdown is on. 10 days left. Who shall rise to the occasion?
> 
> Overlord Genghis (Khan)"

**Comment by Deceth** [comment]
This is not a small feature in its current description. Realistically, the following can be done:

1 - When you open Battle City, an icon with an overlapping # is displayed in your system tray. The number would be the total amount of players online. Players could open Battle City, minimize it, do whatever else they want to do, with the number of players in-game constantly in sight. 

A popup for every player logging on would annoy everyone into closing the game, so possibly at pre-defined intervals such as 10 players, 15 players, 20, etc.. 

Of course, actually coding in all of this, would easily be a 2-3 week full-time project. 

2 - Code something to have the game communicate the total players online with the website. Then code some php to display this somewhere on the website, or create a flash application that can generate popup messages like the current toolbar in the footer.

## 

Actually coding a separate program would be an even bigger project than any option above, and actually distributing this to players to make it useful would be another issue. Better to have the tool be the game itself, or the website.

**Comment by Rushing** [comment]
'When you open Battle City, an icon with an overlapping # is displayed is your system tray. The number would be the total amount of players online. Players could open Battle City, minimize it, do whatever else they want to do, with the number of players in-game constantly in sight.'
This sounds kinda like what I've always hoped for Battlecity to have, how long would it take for just this one part of the idea? It doesn't seem to be that big of a job. O.o

**Comment by Deceth** [comment]
If I were asked to tackle such a job, I would estimate 3 weeks of full-time work. Thats 113 working hours. I have approximately 2 hours of free time per week, so this would take me 56.5 weeks which is 1 year and 1 month. If I started today, I could be done by May 2011.

Of course, in my experience, when I say something will take 3 weeks, multiply it by about 3 to get the real time. So the end result would be a 9 week job or 3 years and 3 months.

**Comment by Rushing** [comment]
...Wow.
I don't really understand how it could take that long, surely you could just use the games database to store who is online add simply update the mini icon (1-99) with how many filled out entries.
And just make B/C show a mini icon in the system tray.

> .<

**Comment by Deceth** [comment]
Yup, that's about it. I challenge you to do this in less than 3 weeks =p

**Comment by Rushing** [comment]
Surely it's simply a matter of using PHP to take the number of entries on the database, then translate it back to C for updating the mini icon.

**Comment by rileyw** [comment]
Was much progress made on this issue?
